src/dbm/dbm_matrix.c

```diff
@@ -34,21 +35,22 @@ void dbm_create(dbm_matrix_t **matrix_out, dbm_distribution_t *dist,
   matrix->dist = dist;
 
   size_t size = (strlen(name) + 1) * sizeof(char);
-  matrix->name = malloc(size);
+  matrix->name = (char *)malloc(size);
   memcpy(matrix->name, name, size);
 
   matrix->nrows = nrows;
   matrix->ncols = ncols;
 
   size = nrows * sizeof(int);
-  matrix->row_sizes = malloc(size);
+  matrix->row_sizes = (int *)malloc(size);
   memcpy(matrix->row_sizes, row_sizes, size);
 
   size = ncols * sizeof(int);
-  matrix->col_sizes = malloc(size);
+  matrix->col_sizes = (int *)malloc(size);
   memcpy(matrix->col_sizes, col_sizes, size);
 
-  matrix->shards = malloc(dbm_get_num_shards(matrix) * sizeof(dbm_shard_t));
+  matrix->shards =
+      (dbm_shard_t *)malloc(dbm_get_num_shards(matrix) * sizeof(dbm_shard_t));
```

Comment:
Is there an advantage to adding these casts? Since the type is already in the `sizeof`, I usually opted for the more compact and readable code.

Reply:
I am not sure actually why it does not warn if casts are not there. Must be some C99 or later thing. Return-type of `malloc` is certainly `void*` which may not promote to any more qualified type without hint (cast).

Reply:
Objective was to not rely on any "permissive" thing.